### PR TITLE
VACMS-6873: show related non-clinical services on top task edit forms…

### LINKED ISSUE #6873: Show related non-clinical services within the Top task edit forms and proofing
## User Story or Problem Statement

As an editor of my Top task page, i can see what related Facility services i already have in my system, from within any of three respective top task edit pages.

## Acceptance Criteria
- [x] A list of facility nonclinical services tagged with Medical records is listed within the VAMC system Medical Records node form and node view
- [x] A list of facility nonclinical services tagged with Register for care is listed within the VAMC system register for care node form and node view
- [x] A list of facility nonclinical services tagged with Billing and insurance is listed within the VAMC system Billing and insurance node form and node view
- [x] Each of these lists shows the following tables
  - [x] Node title with link to node
  - [x] Moderation State
  - [x] Facility 
  - [x] Edit link

### Iterate on `non_clinical_services` view. 

Will need some contextual filter love to list only the Nonclinical service nodes that are associated with that system AND share the same taxonomy term.  The system relationship can be found via field_facility_location -> VAMC facility content type, and from there, via `field_region_page` to VAMC system.  That VAMC system should match the VAMC system in `field_office` on each of the 3 top task content types.

For example, for Sioux Falls Register for care page, we want to show all the non-clincial facility service nodes that have been added to facilities which are in VA Sioux Falls. This is somewhat already. If you add 8135 (The VA Sioux Falls system node ID), it lists all non-clinical services associated with Sioux Falls. But we won’t have 8135 immediately available when editing a Register for care page.  8135 is the value of the `field_office` entity reference on the Register for care node for Sioux Falls. For this, I think we need to add a Views Token Argument to help set the “Provide default value” to 8135, based on the current URL’s field_office:value:NID, or some custom code equivalent. 

### Place within the editorial experience

Placement TBD, probably within a fieldgroup on each of these content type form and view modes. It will be easier to spec placement once the view is built and working

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [x] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [x] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by JayDarnell**
… and proofing

## Description

Closes #6873 

## Testing done

Visual Testing

## Screenshots

VAMC System Register for Care Edit
Screen Shot 2021-12-09 at 10 35 42 PM

VAMC System Register for Care View (Proof)
Screen Shot 2021-12-09 at 10 36 07 PM

VAMC System Billing and Insurance Edit
Screen Shot 2021-12-09 at 10 36 58 PM

VAMC System Billing and Insurance View (Proof)
Screen Shot 2021-12-09 at 10 43 34 PM

VAMC System Medical Records Office Edit
Screen Shot 2021-12-09 at 10 36 29 PM

VAMC System Medical Records Office (Proof)
Screen Shot 2021-12-09 at 10 36 46 PM

## QA steps

As an administrator
For the following steps be sure to choose the same section and VAMC system when creating the nodes (example: VA Alaska Health Care)

**Register for Care**
- [x] Create a new VAMC System Register for Care node (use Alaska for the section and VAMC system)
- [x] Create a new VAMC Facility Non-clinical Service (use Alaska for the section, Register for care for the service and choose any available facility
- [x] Refresh the VAMC System Register for Care node you previously created and verify a view listing the VAMC Facility Non-clinical Service you created is listed
- [x] Click edit for the node and verify the view listing the VAMC Facility Non-clinical Service you created is listed

**Billing and Insurance**
- [x] Create a new VAMC System Billing and Insurance node (use Alaska for the section and VAMC system)
- [x] Create a new VAMC Facility Non-clinical Service (use Alaska for the section, Billing and Insurance for the service and choose any available facility
- [x] Refresh the VAMC System Billing and Insurance node you previously created and verify a view listing the VAMC Facility Non-clinical Service you created is listed
- [x] Click edit for the node and verify the view listing the VAMC Facility Non-clinical Service you created is listed

**Medical Records**
- [x] Create a new VAMC System Medical Records Office node (use Alaska for the section and VAMC system)
- [x] Create a new VAMC Facility Non-clinical Service (use Alaska for the section, Medical Records for the service and choose any available facility
- [x] Refresh the VAMC System Medical Records Office node you previously created and verify a view listing the VAMC Facility Non-clinical Service you created is listed
- [x] Click edit for the node and verify the view listing the VAMC Facility Non-clinical Service you created is listed

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### CODE REVIEW

**Comment by kevwalsh** [comment]
It doesn't say there's a conflict here in github, but i think there's still a conflict, cause [on Tugboat](https://tugboat.vfs.va.gov/619d15e0690c686fc5897dc0) i'm seeing this

VACMS-6873__show_related_non-clinical_services_on_top_task_edit_forms…_«_CMS_Pull_Request_Environments_«_CMS_«_My_Projects_«_Tugboat

**Comment by JayDarnell** [comment]
@kevwalsh I will try to get this rebased today. I'm thinking of writing a patch for the viewfield module to display the view in node edit. There is a setting in this module to set a default and not allow it to be changed. I'd like to add a conditional checkbox that only displays if the default and do not allow edits is selected to then show the chosen view on node edit.

**Comment by JayDarnell** [comment]
This pull request accomplishes the goals outlined in the linked issue. Further iterations will be needed to properly format the contents of the view itself and where it should live within the node edit and node view displays.

File: config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml [hunk 1]
```diff
@@ -22,14 +23,14 @@ dependencies:
     - field_group
     - office_hours
     - telephone
+    - viewfield
 third_party_settings:
   field_group:
     group_edit:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml:30]: Klinger is making a run for it.  You have to add this line back in by hand.

File: config/sync/core.entity_form_display.node.vamc_system_medical_records_offi.default.yml [hunk 2]
```diff
@@ -26,14 +27,14 @@ dependencies:
     - entity_field_fetch
     - field_group
     - telephone
+    - viewfield
 third_party_settings:
   field_group:
     group_edit:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vamc_system_medical_records_offi.default.yml:34]: AWOL

File: config/sync/core.entity_form_display.node.vamc_system_register_for_care.default.yml [hunk 3]
```diff
@@ -18,12 +19,12 @@ dependencies:
     - content_moderation
     - entity_field_fetch
     - field_group
+    - viewfield
 third_party_settings:
   field_group:
     group_edit:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.vamc_system_register_for_care.default.yml:26]: AWOL

File: tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature [hunk 4]
```diff
@@ -4,7 +4,7 @@ Feature: Content model: VAMC Content Type fields
   As a content editor
   I want to have content type fields that reflect my content model.
 
-  @dst @field_type @content_type_fields @dstfields @vamc
+  @dst @field_type @content_type_fields @dstfields @vamc
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature:7]: Need to set your editor to trim trailing spaces on .feature.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Nice work on this issue  and patch @JayDarnell 
 https://www.drupal.org/project/viewfield/issues/3252356

Just a couple things needed to get tests passing and remove trailing spaces.

After tests are passing will run through QA steps.

File: config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml (hunk 1 above)
Comment (JayDarnell) [reply config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml:30]: Adding back in

File: config/sync/core.entity_form_display.node.vamc_system_medical_records_offi.default.yml (hunk 2 above)
Comment (JayDarnell) [reply config/sync/core.entity_form_display.node.vamc_system_medical_records_offi.default.yml:34]: Adding back in

File: config/sync/core.entity_form_display.node.vamc_system_register_for_care.default.yml (hunk 3 above)
Comment (JayDarnell) [reply config/sync/core.entity_form_display.node.vamc_system_register_for_care.default.yml:26]: Adding back in

File: tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature (hunk 4 above)
Comment (JayDarnell) [reply tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature]: I believe I've sorted this at least in the short term.

**Comment by JayDarnell** [comment]
Updated behat tests, readded config items accidentally removed by `lando drush cex`, rebased to remove merge conflicts.

File: tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@api
+@api @jaytest
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_vamc_content_type_fields.feature:1]: No Jay

**Review by swirtSJW — APPROVED** [review]
Code looks good.  
Node view 
image


Node edit 
image

No notices or anything in watchdog.

Nice work Jay.

**Comment by swirtSJW** [comment]
image

**Pull request merged**